Dockerfile

```diff
@@ -56,4 +55,4 @@ USER logstash
 WORKDIR /opt/logstash
 
 LABEL retention="prune"
-
+RUN ./gradlew wrapper --warning-mode all
```

Comment:
should we keep it like this? For development, having the wrapper task at end will mean it's not cached if anything's modified

Reply:
I put this here to avoid a deprecation warning during the running of `./gradlew wrapper` - the requirement for `7.x` of gradle is that a `settings.gradle` file is in place for running any gradle command:

```
Starting a Gradle Daemon (subsequent builds will be faster)
Executing Gradle tasks as part of an undefined build has been deprecated. This will fail with an error in Gradle 7.0. Consult the upgrading guide for further information: https://docs.gradle.org/6.3/userguide/upgrading_version_5.html#executing_gradle_without_a_settings_file_has_been_deprecated
```

Reply:
could we copy the settings file after line https://github.com/elastic/logstash/pull/11742/files#diff-3254677a7917c6c01f55212f86c57fbfL24 ?

Reply:
I'd rather do the extra steps here in the Dockerfile and reduce cache invalidations. it's only two workdir calls that won't increase the size of the image and will improve the dev cycle speed